### DISCUSSION TITLE
Provisioning profile (...) doesn't include the com.apple.developer.proximity-reader.payment.acceptance entitlement error

### DISCUSSION THREAD

**Discussion by rodrigoreyes79** [codemagic-ci-cd/codemagic-docs#2006] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Trigger a new "Default Workflow (iOS Only)" build using tag v2.0.0+31-dev

### Expected results

A new successful build since profile "LinvioPay Mobile ios_app_adhoc 1693339236" does include the com.apple.developer.proximity-reader.payment.acceptance entitlement according to what we can see from the Apple Developer Portal:

![SCR-20230829-nmqo](https://github.com/codemagic-ci-cd/codemagic-docs/assets/3924426/184e2cd4-f3ae-4b29-b9c5-34bb0086e88c)

### Actual results

In the "Building iOS" step, the following error shows up: `
Error (Xcode): Provisioning profile "LinvioPay Mobile ios_app_adhoc 1693339236" doesn't include the com.apple.developer.proximity-reader.payment.acceptance entitlement. Profile qualification is using entitlement definitions that may be out of date. Connect to network to update.`

### Build id (optional)

64ee4e191ca0a6914040fbdd

**Comment by dtrdic** [comment]
hey @rodrigoreyes79, can you try to re-download the provisioning profile from apple and upload it to Codemagic? From the error message it looks like you are still using the provisioning profile which doesn't include the right entitlement

**Reply by dtrdic** [reply]
hey @rodrigoreyes79, have you managed to fix the issue?

**Reply by rodrigoreyes79** [reply]
Hi @dtrdic. Unfortunately, I haven't been able to fix the issue. Followed your suggestions but it looks as if the problem lies in the fact that I am using Firebase App Distribution for testing both on iOS and Android. Given the fact that we received the entitlement approval from Apple exclusively for testing, we can't build for Ad-hoc distribution. Now, I am thinking that maybe the way I am building it can be changed to avoid creating an Ad-hoc distributable package. What would be your suggestion if all we need is to be able to test it on a handful of devices? What do I need to change here (or somewhere else)?

![SCR-20230905-jpmi](https://github.com/codemagic-ci-cd/codemagic-docs/assets/3924426/7fec9da7-2840-4730-b101-dd00ea2d9d68)

Btw, thank you for your help so far :)

**Reply by dtrdic** [reply]
hey @rodrigoreyes79 yes, unfortunately, you will need the Ad-hoc profile for the Firebase distribution.. what if you try distributing to [App store](https://docs.codemagic.io/flutter-publishing/publishing-to-app-store/)?

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
